pandas/core/base.py

```diff
@@ -440,7 +440,9 @@ def nunique(self, dropna=True):
         -------
         nunique : int
         """
-        return len(self.value_counts(dropna=dropna))
+        if dropna:
+            return len(set(self.unique()) - {None})
```

Comment:
This is probably failing because we usually represent NA in pandas as `np.nan`, not `None`